### PR TITLE
bt-208: add logic to onboarding step 4

### LINKED ISSUE #208: Add logic to Onboarding Step 4
**User story:**
As a talent, I want to be able to fill up a profile with my contact information, so that potential employers can access this info after contact happened.

**Acceptance criteria:**
- User should be able to open the steps list with burger menu on the left or with a right swipe 
- The registration step “STEP 04/05” is placed in the upper right corner on the same line with the title CV and contacts
- The fourth step must include input fields for Full name, Phone number, LinkedIn profile, Choose photo, and CV
- The Full name, Phone number, LinkedIn profile, and CV are required and have “*”
- Text inputs should have a placeholders
- Page must contain Choose photo button to add a photo
- When talent clicks on “Choose photo” button should be 2 options: launch gallery or take picture from camera
- Allowed image file is <5MB and could be only jpeg, png, jpg
- The Full name field should be between 3 and 50 characters in length and accept A-Z, a-z, (.), ('), (-), spaces
- The Phone number must start with "+" symbol at the beginning and allow up to 10 symbols total, digits 0-9 only
- Any other symbols in text fields or unsupported length, format should trigger the validation message appearance
- LinkedIn profile field accepts only links in the format of: https://www.linkedin.com/in/ + >0 symbols
- LinkedIn profile field length should be 30-250 chars and accept only: A-Z, a-z, 0-9, special chars
- Page must contain Choose file button to add a CV file to add a file from the device
- Allowed types for CV file should be docx, doc, pdf, max 5MB size
- Talent should be able to back to the previous step Skills and projects by clicking on the Back button
- Talent should be able to go to the next step Preview by clicking on the Next button. Only if no validation errors are present.
- Talent's information should be saved in the fields when switching between steps

### PR DESCRIPTION
**Pull request by illia-dobro**
cv

### CODE REVIEW

File: mobile/src/bundles/common/components/file-picker/file-picker.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+import { Button, Text } from '~/bundles/common/components/components';
+import {
+    ButtonType,
+    IconName,
+    TextCategory,
+} from '~/bundles/common/enums/enums';
+import { useFormController } from '~/bundles/common/hooks/hooks';
+import { globalStyles } from '~/bundles/common/styles/global-styles';
+import { fileSizeValidation } from '~/bundles/talent/components/cv-and-contacts-form/helpers/file-size-validation';
```
Comment (katerynakostikova) [inline mobile/src/bundles/common/components/file-picker/file-picker.tsx]:
```suggestion
import { fileSizeValidation } from '~/bundles/talent/helpers';
```

File: mobile/src/bundles/common/components/file-picker/file-picker.tsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+} from 'react-native-document-picker';
+
+import { Button, Text } from '~/bundles/common/components/components';
+import {
+    ButtonType,
+    IconName,
+    TextCategory,
+} from '~/bundles/common/enums/enums';
+import { useFormController } from '~/bundles/common/hooks/hooks';
+import { globalStyles } from '~/bundles/common/styles/global-styles';
```
Comment (katerynakostikova) [inline mobile/src/bundles/common/components/file-picker/file-picker.tsx]:
```suggestion
import { globalStyles } from '~/bundles/common/styles/styles';
```

File: mobile/src/bundles/common/components/file-picker/file-picker.tsx [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    name,
+    setError,
+}: FilePickerProperties<T>): JSX.Element => {
+    const { field } = useFormController({ name, control });
+
+    const { value, onChange } = field;
+
+    const handleError = (error: unknown): void => {
+        if (isCancel(error)) {
+            notifications.showError({ title: 'cancelled' });
```
Comment (katerynakostikova) [inline mobile/src/bundles/common/components/file-picker/file-picker.tsx]: let's skip toast for cancelled

File: mobile/src/bundles/common/components/file-picker/file-picker.tsx [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+        } else if (isInProgress(error)) {
+            notifications.showError({
+                title: 'multiple pickers were opened, only the last will be considered',
+            });
```
Comment (katerynakostikova) [inline mobile/src/bundles/common/components/file-picker/file-picker.tsx]: what does it mean

File: mobile/src/bundles/common/components/file-picker/file-picker.tsx [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+            const errorMessage = getErrorMessage(error);
+            setError(name, {
+                message: errorMessage,
+            });
+        }
+    };
+
+    const handleDocumentPick = (): void => {
+        DocumentPicker.pick({
+            type: [types.doc, types.docx, types.pdf],
```
Comment (katerynakostikova) [inline mobile/src/bundles/common/components/file-picker/file-picker.tsx]: let's add supported types to constants

File: mobile/src/bundles/common/components/file-picker/file-picker.tsx [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+                    fileSizeValidation(name, document.size);
+                }
+
+                onChange({
+                    name: document.name,
+                    size: document.size,
+                    uri: document.uri,
+                });
+            })
+            .catch(handleError);
```
Comment (katerynakostikova) [inline mobile/src/bundles/common/components/file-picker/file-picker.tsx]: also please rewrite this with async await approach

File: mobile/src/bundles/talent/components/avatar-picker/avatar-picker.tsx [hunk 7]
```diff
@@ -33,16 +54,23 @@ const AvatarPicker: React.FC<AvatarPickerProperties> = ({
...
                 }
                 const [image] = assets;
-                setAvatar(image.uri ?? uri);
-            } catch (error) {
-                if (error instanceof Error) {
-                    notifications.showError({ title: error.message });
-                    return;
+                if (image.fileSize && image.type) {
+                    fileSizeValidation(name, image.fileSize);
+                    imageTypeValidation(image.type);
```
Comment (katerynakostikova) [inline mobile/src/bundles/talent/components/avatar-picker/avatar-picker.tsx]: let's avoid this if else checks in multiple places and just do this check in helpers

File: mobile/src/bundles/talent/components/cv-and-contacts-form/cv-and-contacts-form.tsx [hunk 8]
```diff
@@ -2,16 +2,18 @@ import React from 'react';
...
     ScrollView,
     Text,
     View,
 } from '~/bundles/common/components/components';
-import { IconName, TextCategory } from '~/bundles/common/enums/enums';
+import { TextCategory } from '~/bundles/common/enums/enums';
 import { useAppForm, useCallback } from '~/bundles/common/hooks/hooks';
 import { globalStyles } from '~/bundles/common/styles/styles';
 import { AvatarPicker } from '~/bundles/talent/components/avatar-picker/avatar-picker';
+import { CvAndContactsFormValidationSchema } from '~/bundles/talent/validation-schemas/validation-schemas';
```
Comment (katerynakostikova) [inline mobile/src/bundles/talent/components/cv-and-contacts-form/cv-and-contacts-form.tsx:16]: Does the frontend use different validation rules?

File: mobile/src/bundles/talent/components/cv-and-contacts-form/helpers/file-size-validation.ts [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+import { MAX_FILE_SIZE } from '../constants/constants';
+
+const fileSizeValidation = <T extends FieldValues>(
+    name: FieldPath<T>,
+    fileSize: number,
+): void => {
+    if (fileSize > MAX_FILE_SIZE.bytes) {
+        const errorMessage = `Please upload a ${name} smaller than ${MAX_FILE_SIZE.mb} MB.`;
+        throw new Error(errorMessage);
+    }
```
Comment (katerynakostikova) [inline mobile/src/bundles/talent/components/cv-and-contacts-form/helpers/file-size-validation.ts]: let's rewrite this helper as checkIfFileSizeValid and return a boolean

File: mobile/src/bundles/talent/components/cv-and-contacts-form/helpers/image-type-validation.ts [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+
+const imageTypeValidation = (type: string): void => {
+    const typeImage = type.split('/')[lastIndex];
+
+    if (!ACCEPTED_PHOTO_TYPES.includes(typeImage)) {
+        const errorMessage = `Allowed image file could be only ${ACCEPTED_PHOTO_TYPES.join(
+            ', ',
+        )}`;
+        throw new Error(errorMessage);
+    }
```
Comment (katerynakostikova) [inline mobile/src/bundles/talent/components/cv-and-contacts-form/helpers/image-type-validation.ts]: same here

File: mobile/src/bundles/talent/components/cv-and-contacts-form/cv-and-contacts-form.tsx (hunk 8 above)
Comment (illia-dobro) [reply mobile/src/bundles/talent/components/cv-and-contacts-form/cv-and-contacts-form.tsx:16]: The validation was moved from the shared folder to the frontend folder. It is mostly the same, but there might be some differences for CV and image.

File: mobile/src/bundles/common/components/file-picker/constants/constants.ts [hunk 11]
```diff
@@ -0,0 +1,9 @@
+import { types } from 'react-native-document-picker';
+
+const ACCEPTED_DOCUMENT_TYPE = {
+    DOC: types.doc,
+    DOCX: types.docx,
+    PDF: types.pdf,
+};
```
Comment (katerynakostikova) [inline mobile/src/bundles/common/components/file-picker/constants/constants.ts]:
```suggestion
const ACCEPTED_DOCUMENT_TYPES = [types.doc,  types.docx, types.pdf] 
```

File: mobile/src/bundles/talent/components/avatar-picker/avatar-picker.tsx [hunk 12]
```diff
@@ -8,23 +12,40 @@ import {
...
+import { getErrorMessage } from '~/bundles/common/helpers/helpers';
+import {
+    useCallback,
+    useFormController,
+    useState,
+} from '~/bundles/common/hooks/hooks';
 import { globalStyles } from '~/bundles/common/styles/styles';
 import { type AvatarProperties } from '~/bundles/common/types/types';
+import { checkIfFileSizeValid } from '~/bundles/talent/helpers/check-if-file-size-valid';
+import { checkIfImageTypeValid } from '~/bundles/talent/helpers/check-if-image-type-valid';
```
Comment (katerynakostikova) [inline mobile/src/bundles/talent/components/avatar-picker/avatar-picker.tsx]:
```suggestion
import { checkIfFileSizeValid, checkIfImageTypeValid } from '~/bundles/talent/helpers/helpers';
```

**Review by katerynakostikova — APPROVED** [review]

**Review by anogire — APPROVED** [review]

**Comment by katerynakostikova** [comment]
resolve conflicts

**Pull request merged**